src/ApiUtils/include/ApiUtils/EncodedEvent.h

```diff
@@ -13,82 +13,6 @@
 
 namespace orbit_api {
 
-constexpr uint8_t kVersion = 1;
-
-enum EventType : uint8_t {
-  kNone = 0,
-  kScopeStart = 1,
-  kScopeStop = 2,
-  kScopeStartAsync = 3,
-  kScopeStopAsync = 4,
-  kTrackInt = 5,
-  kTrackInt64 = 6,
-  kTrackUint = 7,
-  kTrackUint64 = 8,
-  kTrackFloat = 9,
-  kTrackDouble = 10,
-  kString = 11,
-};
-
-constexpr size_t kMaxEventStringSize = 34;
-struct Event {
-  uint8_t version;                 // 1
-  uint8_t type;                    // 1
-  char name[kMaxEventStringSize];  // 34
-  orbit_api_color color;           // 4
-  uint64_t data;                   // 8
-};
-
-// EncodedEvent is used for encoding an Orbit API event into the 6 integer register of the Linux
-// x64 ABI. This is useful for the version of the manual instrumentation API that relies on uprobes.
-union EncodedEvent {
-  EncodedEvent() = default;
-  EncodedEvent(orbit_api::EventType type, const char* name = nullptr, uint64_t data = 0,
-               orbit_api_color color = kOrbitColorAuto) {
-    static_assert(sizeof(EncodedEvent) == 48, "orbit_api::EncodedEvent should be 48 bytes.");
-    static_assert(sizeof(Event) == 48, "orbit_api::Event should be 48 bytes.");
-    event.version = kVersion;
-    event.type = static_cast<uint8_t>(type);
-    memset(event.name, 0, kMaxEventStringSize);
-    if (name != nullptr) {
-      std::strncpy(event.name, name, kMaxEventStringSize - 1);
-      event.name[kMaxEventStringSize - 1] = 0;
-    }
-    event.data = data;
-    event.color = color;
-  }
-
-  EncodedEvent(uint64_t a0, uint64_t a1, uint64_t a2, uint64_t a3, uint64_t a4, uint64_t a5) {
-    args[0] = a0;
-    args[1] = a1;
-    args[2] = a2;
-    args[3] = a3;
-    args[4] = a4;
-    args[5] = a5;
-  }
-  orbit_api::EventType Type() const { return static_cast<orbit_api::EventType>(event.type); }
-
-  Event event;
-  uint64_t args[6];
-};
-
-// ApiEvent is used for the version of manual instrumentation API that relies on the side channel.
-// It reuses existing EncodedEvent logic but adds information otherwise retrieved through uprobes.
-struct ApiEvent {
-  ApiEvent() = default;
-  ApiEvent(uint32_t pid, uint32_t tid, uint64_t timestamp_ns, orbit_api::EventType type,
-           const char* name = nullptr, uint64_t data = 0, orbit_api_color color = kOrbitColorAuto)
-      : encoded_event(type, name, data, color), pid(pid), tid(tid), timestamp_ns(timestamp_ns) {
-    static_assert(sizeof(ApiEvent) == 64, "orbit_api::ApiEvent should be 64 bytes.");
-  }
-  orbit_api::EventType Type() const { return encoded_event.Type(); }
-
-  EncodedEvent encoded_event;
-  uint32_t pid;
-  uint32_t tid;
-  uint64_t timestamp_ns;
-};
-
 template <typename Dest, typename Source>
 inline Dest Encode(const Source& source) {
```

Comment:
`Encode` and `Decode` are now only used in `CaptureEventProcessorForListener` and in `OrbitGl` for the memory events (which I remember disagreeing with...), right? I think the remaining of this file should be moved, maybe in `CaptureClient` if the dependencies allow it? If this is something you don't feel like tackling, a big TODO and explanation of the fact that they are out of place seems appropriate. (Of course the cleaner solution would be to get rid of them and rework the representation of the memory events, but that's a different story.

Reply:
I will actually remove the Encode/Decode completely in  one of my next PRs, as this is a bit of a bigger thing and kinda unrelated to this here, I'd keep this PR as it is.

Should I still add a TODO?

Reply:
If you are already planning to tackle this, a short TODO or no TODO are fine.

Reply:
Done.